### PR TITLE
Bug 1382153, Added more details to the Service Serving Certificate Secrets section

### PR DESCRIPTION
**Pull request by ahardin-rh**
https://bugzilla.redhat.com/show_bug.cgi?id=1382153

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@adellape @deads2k PTAL

File: dev_guide/secrets.adoc [hunk 1]
```diff
@@ -148,11 +148,13 @@ serving certificate/key pair into a secret in your namespace. To do this, set
...
 available, your pod will run. The certificate will be good for the internal
-service DNS name, `<service.name>.<service.namespace>.svc`. The certificate
-and key are in PEM format, stored in `tls.crt` and `tls.key` respectively.
-They are regenerated upon expiration. View the expiration date in the
-`service.alpha.openshift.io/expiry` annotation on the secret, which is in
-RFC3339 format.
+service DNS name, `<service.name>.<service.namespace>.svc`. .
+
+The certificate and key are in PEM format, stored in `tls.crt` and `tls.key`
+respectively. They are regenerated upon expiration. The certificate/key pair is
```
Comment (deads2k) [inline dev_guide/secrets.adoc]: "They are regenerated upon expiration." is unnecessary now that it is "when it gets close"

File: dev_guide/secrets.adoc [hunk 2]
```diff
@@ -167,6 +169,12 @@ internal DNS names), by using the CA bundle in the
 *_/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt_* file that is
 automatically mounted in their pod.
 
+The signature algorithm for this feature is `x509.SHA256WithRSA`. This feature
+is intended to support complex middleware applications that need out-of-the-box
+certificates. It has the same settings as the server certificates generated by
+the administrator tooling for nodes and masters. To rotate, delete the
```
Comment (deads2k) [inline dev_guide/secrets.adoc]: "To manually rotate, delete the generated secret."

**Comment by deads2k** [comment]
just minor comments.

File: dev_guide/secrets.adoc [hunk 3]
```diff
@@ -148,11 +148,13 @@ serving certificate/key pair into a secret in your namespace. To do this, set
 the `service.alpha.openshift.io/serving-cert-secret-name` to the name you want
 to use for your secret. Then, your *PodSpec* can mount that secret. When it is
 available, your pod will run. The certificate will be good for the internal
-service DNS name, `<service.name>.<service.namespace>.svc`. The certificate
-and key are in PEM format, stored in `tls.crt` and `tls.key` respectively.
-They are regenerated upon expiration. View the expiration date in the
-`service.alpha.openshift.io/expiry` annotation on the secret, which is in
-RFC3339 format.
+service DNS name, `<service.name>.<service.namespace>.svc`. .
```
Comment (adellape) [inline dev_guide/secrets.adoc]: An extra `.` here.

File: dev_guide/secrets.adoc [hunk 4]
```diff
@@ -167,6 +169,12 @@ internal DNS names), by using the CA bundle in the
 *_/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt_* file that is
 automatically mounted in their pod.
 
+The signature algorithm for this feature is `x509.SHA256WithRSA`. This feature
+is intended to support complex middleware applications that need out-of-the-box
```
Comment (adellape) [inline dev_guide/secrets.adoc]: > This feature is intended to support complex middleware applications that need out-of-the-box certificates. It has the same settings as the server certificates generated by the administrator tooling for nodes and masters.

These sentences could potentially go up into the first paragraph, either before or after the first sentence there? Feels a little strange to be explaining the use case at the very end, after having gotten into the technical weeds.

Comment (adellape) [inline dev_guide/secrets.adoc]: Also maybe here:

s/This feature is/Service serving certificate secrets are/

**Comment by ahardin-rh** [comment]
@deads2k @adellape Thanks! This is updated.

**Review by adellape — APPROVED** [review]

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../dev_guide/secrets.adoc#dev-guide-secrets[Secrets]
|Added more details to the xref:../dev_guide/secrets.adoc#service-serving-certificate-secrets[Service Serving Certificate Secrets] section.
%

**Pull request merged**